### PR TITLE
PCH Title Suggestions: Update Title Suggestions panel UI

### PR DESCRIPTION
**Pull request by vaurdan**
## Description
Updates the Title Suggetions Panel UI to improve user experience. 

This PR adds a new Telemetry event, as described below:

| Event Name                                           | Description                                                      |
|------------------------------------------------------|------------------------------------------------------------------|
| `title_suggestions_pinned_toggle`        | Fires when the pinned titles panel is collapsed or uncollapsed  |

## Motivation and context
* Improve the UX of the Parse.ly Content Helper
* This PR is to be merged into #2238 

## How has this been tested?
Tested locally with all the new changes validated.

### CODE REVIEW

File: src/content-helper/common/components/persona-selector/component.tsx [hunk 1]
```diff
@@ -11,63 +11,57 @@ import {
 import { useDebounce } from '@wordpress/compose';
 import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Icon, chevronRight } from '@wordpress/icons';
+import { Icon, edit } from '@wordpress/icons';
+import { MoreArrow } from '../../icons/more-arrow';
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/component.tsx]: Let's move line 15 under internal dependencies.

File: src/content-helper/common/components/persona-selector/component.tsx [hunk 2]
```diff
@@ -11,63 +11,57 @@ import {
...
 
 /**
  * Internal dependencies
  */
-import { PersonIcon } from '../../icons/person-icon';
+type PersonaMetadata = {
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/component.tsx:26]: Let's add a comment to this type as we've done for others.

File: src/content-helper/common/components/persona-selector/component.tsx [hunk 3]
```diff
@@ -77,13 +71,13 @@ type FixedPersonaProp = keyof typeof PARSELY_PERSONAS;
...
- * Returns the label for a given persona.
+ * Returns the value for a given persona.
  *
  * @since 3.13.0
  *
- * @param {PersonaProp} persona The persona to get the label for.
+ * @param {PersonaProp} persona The persona to get the value for.
  *
- * @return {string} The label for the given persona.
+ * @return {string} The value for the given persona.
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/component.tsx:86]: Comments here have been updated with `value` but the const below them is named `getLabel`. I don't know if this is correct, just flagging.

File: src/content-helper/common/components/persona-selector/style.scss [hunk 4]
```diff
@@ -3,9 +3,23 @@
 
 .parsely-persona-selector-dropdown {
 	width: 100%;
-	margin-bottom: to_rem(10px);
 	overflow-wrap: break-word;
 	word-break: break-word;
+	height: 36px;
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/style.scss]: Do we need `to_rem()` here?

File: src/content-helper/common/components/persona-selector/style.scss [hunk 5]
```diff
@@ -3,9 +3,23 @@
...
+	border-radius: 2px;
+	border: 1px solid var(--Gutenberg-Gray-600, #949494);
+	background: var(--Gutenberg-White, #fff);
+
+	.components-dropdown-menu__toggle.has-icon svg:nth-child(1) {
+		display: none;
+	}
+
+	button {
+		height: 33px;
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/style.scss]: Do we need `to_rem()` here?

File: src/content-helper/common/components/persona-selector/style.scss [hunk 6]
```diff
@@ -21,23 +35,26 @@
...
+.parsely-tone-selector-custom {
+	width: 100%;
 
-		&:focus {
-			outline: 3px solid transparent;
-			box-shadow: 0 0 0 2px var(--parsely-green);
-		}
+	.components-base-control__field input {
+		display: flex;
+		height: 40px;
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/style.scss]: Do we need `to_rem()` here?

File: src/content-helper/common/components/tone-selector/component.tsx [hunk 7]
```diff
@@ -11,58 +11,55 @@ import {
 import { useDebounce } from '@wordpress/compose';
 import { useState } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
-import { Icon, chevronRight, megaphone } from '@wordpress/icons';
+import { Icon, edit } from '@wordpress/icons';
+import { MoreArrow } from '../../icons/more-arrow';
+import { PARSELY_PERSONAS } from '../persona-selector';
```
Comment (acicovic) [inline src/content-helper/common/components/tone-selector/component.tsx]: Let's move these to an internal dependencies section.

File: src/content-helper/common/components/tone-selector/component.tsx [hunk 8]
```diff
@@ -11,58 +11,55 @@ import {
...
+
+type ToneMetadata = {
```
Comment (acicovic) [inline src/content-helper/common/components/tone-selector/component.tsx:26]: Let's add a comment to this type as we've done for others.

File: src/content-helper/common/components/tone-selector/component.tsx [hunk 9]
```diff
@@ -72,13 +69,13 @@ type FixedToneProp = keyof typeof PARSELY_TONES;
...
- * Returns the label for a given tone.
+ * Returns the value for a given tone.
  *
  * @since 3.13.0
  *
- * @param {ToneProp} tone The tone to get the label for.
+ * @param {ToneProp} tone The tone to get the value for.
  *
- * @return {string} The label for the given tone.
+ * @return {string} The value for the given tone.
```
Comment (acicovic) [inline src/content-helper/common/components/tone-selector/component.tsx:86]: Same thing I reported in a previous file about value and label. Ignore if this was done on purpose.

File: src/content-helper/common/components/tone-selector/style.scss [hunk 10]
```diff
@@ -2,9 +2,23 @@
 
 .parsely-tone-selector-dropdown {
 	width: 100%;
-	margin-bottom: to_rem(10px);
 	overflow-wrap: break-word;
 	word-break: break-word;
+	height: 36px;
```
Comment (acicovic) [inline src/content-helper/common/components/tone-selector/style.scss]: Comments about possibly needing `to_rem()` apply to this file as well.

File: src/content-helper/common/settings/types/index.ts [hunk 11]
```diff
@@ -1,14 +1,20 @@
...
+} from './sidebar-settings';
 import type { TopPostsSettings } from './top-posts-settings';
 
 /**
  * Export the settings types.
  */
 export type {
 	SidebarSettings,
+	TitleSuggestionsSettings, // Part of SidebarSettings type
+	PerformanceStatsSettings, // Part of SidebarSettings type
```
Comment (acicovic) [inline src/content-helper/common/settings/types/index.ts]:
```suggestion
	TitleSuggestionsSettings, // Part of SidebarSettings type.
	PerformanceStatsSettings, // Part of SidebarSettings type.
```

File: src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx [hunk 12]
```diff
@@ -0,0 +1,69 @@
+/**
+ * WordPress dependencies
+ */
+import { Panel, PanelBody } from '@wordpress/components';
+import { useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { Telemetry } from '../../../js/telemetry/telemetry';
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx]: Line 7 should go to internal dependencies.

File: src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx [hunk 13]
```diff
@@ -0,0 +1,69 @@
...
+				{ pinnedTitles.map( ( title ) => (
+					<TitleSuggestion
+						key={ title.title }
+						title={ title }
+						type={ TitleType.PostTitle }
+					/>
+				) ) }
+			</PanelBody>
+		</Panel>
+
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx]:
```suggestion
```
Empty line removal

File: src/content-helper/editor-sidebar/title-suggestions/component.tsx [hunk 14]
```diff
@@ -117,153 +134,157 @@ export const TitleSuggestionsPanel = (): JSX.Element => {
 		}
 	};
 
-	const saveTitleOnClickHandler = async () => {
+	/**
+	 * Handle the accepted title changing, and apply the accepted title to the post.
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component.tsx]:
```suggestion
	 * Handles the accepted title changing, and apply the accepted title to the post.
```
I've always treated `useEffect` comments as function comments, but we can change this if you think it shouldn't be the case.

File: src/content-helper/editor-sidebar/title-suggestions/component.tsx [hunk 15]
```diff
@@ -117,153 +134,157 @@ export const TitleSuggestionsPanel = (): JSX.Element => {
...
-				<TitleSuggestion
-					key={ title.id }
-					title={ title }
-					type={ TitleType.PostTitle } // Specify that the title is a post title.
-				/>
-			) ) }
-		</div>
-	);
+	/**
+	 * Display a snackbar notification when an error occurs.
```
Comment (acicovic) [inline src/content-helper/editor-sidebar/title-suggestions/component.tsx]:
```suggestion
	 * Displays a snackbar notification when an error occurs.
```

**Review by acicovic — COMMENTED** [review]
Thank you for this!

I've done a first quick pass and left comments about minor things. I'm planning on doing a second pass once the current feedback is addressed/discussed, and I'll also give this a test try.

File: src/content-helper/common/components/persona-selector/component.tsx (hunk 1 above)
Comment (vaurdan) [reply src/content-helper/common/components/persona-selector/component.tsx]: Oops! Good catch.

File: src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx (hunk 12 above)
Comment (vaurdan) [reply src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx]: PHPStorm import optimizer keeps messing this out... Thanks for catching it!

Comment (acicovic) [reply src/content-helper/editor-sidebar/title-suggestions/component-pinned.tsx]: No worries.

File: src/content-helper/common/components/persona-selector/component.tsx [hunk 16]
```diff
@@ -11,63 +11,63 @@ import {
...
+ */
+type PersonaMetadata = {
+	label: string,
+	icon?: JSX.Element,
+};
 
 /**
  * List of the available personas.
- * Each persona has a label and an emoji.
+ * Each persona has a value and an icon.
```
Comment (acicovic) [inline src/content-helper/common/components/persona-selector/component.tsx]: Is this meant to be `a label and an icon`?

File: src/content-helper/common/components/tone-selector/component.tsx [hunk 17]
```diff
@@ -11,58 +11,63 @@ import {
...
+ */
+type ToneMetadata = {
+	label: string,
+	icon?: JSX.Element,
+};
 
 /**
  * List of the available tones.
- * Each tone has a label and an emoji.
+ * Each tone has a value and an icon.
```
Comment (acicovic) [inline src/content-helper/common/components/tone-selector/component.tsx]: Is this meant to be `a label and an icon`?

File: src/content-helper/common/components/persona-selector/component.tsx (hunk 16 above)
Comment (acicovic) [reply src/content-helper/common/components/persona-selector/component.tsx]: Or `a label and an optional icon`.

**Comment by acicovic** [comment]
I've noticed that when migrating from `develop` to this branch, (flow that will be experienced from current PCH users) Title Suggestion settings are shown like this:

image

It would be better to have the experience of new installations (no settings exist), which works as expected:

image

**Comment by acicovic** [comment]
Some additional feedback:
- If we can, it would be nice to set the focus to the respective text control when choosing Custom Tone/Persona. Makes UX faster and easier.
- Selecting a predefined Tone/Persona empties the custom Tone/Persona value. I think we should keep the custom values, but let me know if this was done on purpose.

**Comment by vaurdan** [comment]
@acicovic thanks for your feedback! 

> I've noticed that when migrating from develop to this branch, (flow that will be experienced from current PCH users) Title Suggestion settings are shown like this:

Good catch! I have addressed this and the settings are now valid when migrating to the new version.

> If we can, it would be nice to set the focus to the respective text control when choosing Custom Tone/Persona. Makes UX faster and easier.

Nice idea! I have implemented this and the focus now changes to the input text when setting a custom tone or persona.

> Selecting a predefined Tone/Persona empties the custom Tone/Persona value. I think we should keep the custom values, but let me know if this was done on purpose.

This behaviour is the same as before the UI improvements. The way the Title Suggestions store - and the respective settings store - works is by using the tone and persona fields for native values, or custom ones. So, in order to implement this suggestion, we would need to add a new fields to the Title Suggestion store (and Settings store), to store both the custom tone and the custom persona. It's definitely doable and not too complicated, but want to confirm with you first if we should move this forward either way.

**Comment by acicovic** [comment]
> Selecting a predefined Tone/Persona empties the custom Tone/Persona value. I think we should keep the custom values, but let me know if this was done on purpose.
>> This behaviour is the same as before the UI improvements.

Strange. I thought this is the way it worked before but I've tested it and you're right. No need to do anything more in this PR.

**Review by acicovic — APPROVED** [review]
Very exciting work, thank you! 🙂

**Pull request merged**